src/main/java/com/internet/shop/dao/jdbc/ShoppingCartJdbcImpl.java

```diff
@@ -0,0 +1,173 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartJdbcImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO shopping_carts (user_id) VALUES (?);";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, shoppingCart.getUserId());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            while (generatedKeys.next()) {
+                long shoppingCartId = generatedKeys.getLong(1);
+                shoppingCart.setId(shoppingCartId);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("It is impossible to create "
+                    + "a shopping cart for user with id: "
+                    + shoppingCart.getUserId(), e);
+        }
+        insertDataToShoppingCartsProducts(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM shopping_carts WHERE cart_id = ? AND deleted = FALSE";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                ShoppingCart shoppingCart = createShoppingCartFromResultSet(resultSet);
+                shoppingCart.setProducts(extractProductsForShoppingCart(shoppingCart.getId()));
```

Comment:
same here/ You can move it after catch block or pass connection object to method